[HOLD] DAG to sync vendor data from folio and deprecate removed interfaces

lwrubel opened the pull request:
Requires tests and more testing, as well as confirming UI makes sense for deprecated interfaces.

jermnelson commented on libsys_airflow/plugins/vendor/sync.py:
```diff
@@ -0,0 +1,147 @@
+def sync_data(folio_client):
+    organizations = folio_client.organizations
+    acq_names = _get_acquisitions_unit_names(
+        set([org["acqUnitIds"][0] for org in organizations]), folio_client
```
Unrelated to this PR, I was seeing an IndexError in similar code when an organization did not have an assigned Acquisitions Unit Id with the `org["acqUnitIds"]` returned an empty list. This may not be issue if all of the organizations have at least one acquisition unit. You could add a guard clause to this list comprehension like 
```
set([org["acqUnitIds"][0] for org in organizations if len(org["acqUnitIds"]) > 0)

lwrubel replied:
I see in the models that:
`acquisitions_unit_from_folio = Column(String(36), unique=False, nullable=False)`

Should this field be nullable?

jermnelson replied:
Since acquisitions Unit Ids are not in the required fields for [Organizations ](https://github.com/folio-org/acq-models/blob/505ff2e91f7d5f0e1fc164ad959d4af7e8d76900/mod-orgs/schemas/organization.json#L430-L433 ), I think this field should be nullable. Good catch1

lwrubel replied:
I've added a migration to make `acquisitions_unit_from_folio` nullable.

jermnelson reviewed (changes requested):
Suggestion for a small update to the list comprehension to guard against an empty acqUnitIds list.

justinlittman commented on libsys_airflow/plugins/vendor/sync.py:
```diff
@@ -0,0 +1,147 @@
+            removed = list(
+                set(existing_vendor_interfaces).difference(organization["interfaces"])
+            )
+            for interface_uuid in removed:
```
Interfaces which are upload only (i.e., have a null folio_interface_uuid) should not be removed. I believe with this logic they are.

lwrubel replied:
Thanks for catching. Updated.

lwrubel replied:
Also added a test case to cover sync on upload-only.

lwrubel commented:
Hoping it's less confusing, I flipped the `removed_in_folio` field to a positive and renamed it `assigned_in_folio`. 

![Screenshot 2023-06-06 at 10 20 56 AM](https://github.com/sul-dlss/libsys-airflow/assets/1619369/c9012441-8931-4b6f-aed3-3fa8ce6d9499)

Unassign/assign is terminology used in the FOLIO UI. Upload-only interfaces will also have `assigned_in_folio` be False.

@justinlittman I added logic to avoid changing the status of upload-only interfaces when the sync DAG runs.

lwrubel commented on libsys_airflow/plugins/vendor/sync.py:
```diff
@@ -0,0 +1,155 @@
+            )
+            for interface_uuid in unassigned:
+                if interface_uuid is not None:  # do not unassign upload-only interfaces
+                    unassigned_interface = (
```
Fixing logic to keep upload-interfaces from getting reset.

justinlittman reviewed (approved):
@lwrubel There have been some recent db migrations added. Does this need to be adjusted?
